Structure_Engine/Create/SectionProperties/LongitudinalReinforcement.cs

```diff
@@ -0,0 +1,193 @@
+                 "Starts by fitting in as many points as possible in a layer towards the top of the section, then generates a new one and repeats.")]
+        [InputFromProperty("diameter")]
+        [Input("area", "Total minimum required area of bottom reinforcement. Will be used to calculate required number of bars, based on their diameter, hence the resulting area can get higher the inout value.", typeof(Area))]
+        [Input("spacing", "Minimum spacing allowed between any two rebars.")]
+        [InputFromProperty("startLocation")]
+        [InputFromProperty("endLocation")]
+        [Input("material", "Material of the Rebars. If null, a defaul material will be pulled from the Datasets.")]
+        [Output("reinforcement", "The created Reinforcement to be applied to a ConcreteSection.")]
+        public static LongitudinalReinforcement TopReinforcement(double diameter, double area, double spacing, double startLocation = 0, double endLocation = 1, IMaterialFragment material = null)
+        {
+            int numberOfBars = (int)Math.Ceiling(area / (diameter * diameter * Math.PI / 4));
+            return MultiLinearReinforcement(diameter, numberOfBars, spacing, spacing, Vector.XAxis, 0, ReferencePoint.TopCenter, startLocation, endLocation, material);
+        }
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private static void CheckEndLocations(ref double startLocation, ref double endLocation)
+        {
+            if (startLocation < 0)
+            {
+                startLocation = 0;
+                Reflection.Compute.RecordWarning("Start location need to be larger or equal to 0. To accomodate, the start location has been set to 0.");
+            }
+            else if (startLocation > 1)
+            {
+                startLocation = 1;
+                Reflection.Compute.RecordWarning("Start location need to be smaller or equal to 1. To accomodate, the start location has been set to 1.");
+            }
+
+            if (endLocation < 0)
+            {
+                startLocation = 0;
+                Reflection.Compute.RecordWarning("End location need to be larger or equal to 0. To accomodate, the end location has been set to 0.");
+            }
+            else if (startLocation > 1)
+            {
+                startLocation = 1;
+                Reflection.Compute.RecordWarning("End location need to be smaller or equal to 1. To accomodate, the end location has been set to 1.");
+            }
+
+            if (startLocation > endLocation)
+            {
+                double temp = startLocation;
+                startLocation = endLocation;
+                endLocation = temp;
+                Reflection.Compute.RecordWarning("Start location need to be smaller or equal than the end location. To accomodate, the start and end location have been switched.");
```

Comment:
I love the word 'accommodate' here! "to accommodate your incompetence, I have assumed you meant the following"

Reply:
```suggestion
                Reflection.Compute.RecordWarning("Start location need to be smaller or equal than the end location. To accommodate, the start and end location have been switched.");
```